src/matlab2tikz.m

```diff
@@ -3120,26 +3129,35 @@ function warnFacetedInterp(m2t)
         ['A 3D plot with "shader = faceted interp" is being produced.\n', ...
         'This may produce big and sluggish PDF files.\n', ...
         'See %s and Section 4.6.6 of the pgfplots manual for workarounds.'], ...
-        issueUrl(693, true));
+        issueUrl(m2t, 693, true));
 end
 % ==============================================================================
-function url = issueUrl(number, forOutput)
+function url = issueUrl(m2t, number, forOutput)
```

Comment:
I think both have their merit. 

Yours is without any doubt cleaner in the amount/scope of data that is passed around: not the whole `m2t` god object is needed, just the issue part. :+1:

However, I slightly prefer my approach: in many functions we just pass `m2t` (which is not a best practice, I will admit, but at least we are aware of it: #477 ).  And it's a pattern in the code, so it's not surprising anymore to pass around `m2t`.

Indexing into `m2t` before calling `issueUrl`, just moves that logic (`m2t.about.issue`) to ALL call sites of the function, hence littering those places with unrelated/duplicate logic and possibility for error. Since my approach has that piece of logic in a single place, that cannot happen.

With #477, I think we can certainly have a look at this again. Since we're talking about (global) constants being passed around, I think there are quite a few options to pass those behind the scenes (e.g. wrap those in a function that can be called where needed).